### PR TITLE
WIP: trough for deduplication initial proof-of-concept-ish code

### PR DESCRIPTION
**Pull request by nlevitt**

### CODE REVIEW

File: warcprox/main.py [hunk 1]
```diff
@@ -327,6 +345,7 @@ def ensure_rethinkdb_tables():
     tables. So it's a good idea to use this utility at an early step when
     spinning up a cluster.
     '''
+    raise Exception('adjust my args')
```
Comment (jkafader) [inline warcprox/main.py]: Is this correct?

File: warcprox/trough.py [hunk 2]
```diff
@@ -0,0 +1,238 @@
+'''
```
Comment (jkafader) [inline warcprox/trough.py]: This should be de-duped against the code in the trough repo and the ait5 repo, at some point. We shouldn't have 3 copies of essentially the same code floating around. This seems like the most promising of the three starting points.

File: warcprox/main.py (hunk 1 above)
Comment (nlevitt) [reply warcprox/main.py]: No it's not!

Comment (nlevitt) [reply warcprox/main.py]: Implemented

**Pull request merged**